app/src/main/java/com/simtop/billionbeers/testing_utils/MainCoroutineScopeRule.kt

```diff
@@ -9,28 +9,31 @@ import org.junit.rules.TestWatcher
 import org.junit.runner.Description
 
 @ExperimentalCoroutinesApi
-class MainCoroutineScopeRule(val dispatcher: TestCoroutineDispatcher = TestCoroutineDispatcher()) : TestWatcher() {
+class MainCoroutineScopeRule(
+    val testDispatcher: TestDispatcher = UnconfinedTestDispatcher()
+) : TestWatcher() {
 
-    val testDispatcherProvider = object : CoroutineDispatcherProvider {
-        override val default: CoroutineDispatcher = dispatcher
-        override val io: CoroutineDispatcher = dispatcher
-        override val main: CoroutineDispatcher = dispatcher
-        override val unconfined: CoroutineDispatcher = dispatcher
+    val testScheduler = testDispatcher.scheduler
 
+    val testDispatcherProvider = object : CoroutineDispatcherProvider {
+        override val default: CoroutineDispatcher = testDispatcher
+        override val io: CoroutineDispatcher = testDispatcher
+        override val main: CoroutineDispatcher = testDispatcher
+        override val unconfined: CoroutineDispatcher = testDispatcher
     }
 
     override fun starting(description: Description?) {
         super.starting(description)
-        Dispatchers.setMain(dispatcher)
+        Dispatchers.setMain(testDispatcher)
     }
 
     override fun finished(description: Description?) {
         super.finished(description)
         Dispatchers.resetMain()
-        dispatcher.cleanupTestCoroutines()
     }
 }
 
+//TODO: refactor
 @ExperimentalCoroutinesApi
-fun MainCoroutineScopeRule.runBlocking(block: suspend TestCoroutineScope.() -> Unit) =
-        this.dispatcher.runBlockingTest { block() }
\ No newline at end of file
+fun MainCoroutineScopeRule.runBlocking(block: suspend () -> Unit) =
```

Comment:
I should delete it and use runTest, and also add more tests to see if the Rule works as expected.